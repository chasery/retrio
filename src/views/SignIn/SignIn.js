import React from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import Header from '../../components/Header/Header';
import SignInForm from '../../components/SignInForm/SignInForm';
import './SignIn.css';

function SignIn(props) {
  const location = useLocation();
  const history = useHistory();

  const handleSignInSuccess = () => {
    const destination = (location.state || {}).from || '/racks';

    history.push(destination);
  };

  return (
    <>
      <Header />
      <main role='main'>
        <section className='SignIn'>
          <div className='SignIn__wrapper'>
            <SignInForm onSignInSuccess={handleSignInSuccess} />
          </div>
        </section>
      </main>
    </>
  );
}

export default SignIn;
