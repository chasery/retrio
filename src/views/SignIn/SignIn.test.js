import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignIn from './SignIn';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/sign-in',
  }),
}));

describe('SignIn component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<SignIn />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
