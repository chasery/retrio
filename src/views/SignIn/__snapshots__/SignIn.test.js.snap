// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="SignIn"
    >
      <div
        className="SignIn__wrapper"
      >
        <SignInForm
          onSignInSuccess={[Function]}
        />
      </div>
    </section>
  </main>
</Fragment>
`;
